src/lib/storagevol.cpp

```diff
@@ -55,6 +55,45 @@ QString StorageVol::size()
     return QString();
 }
 
+QString StorageVol::usedby()
+{
+    QString usedbyvm;
+    QString pathdisk = path();
+    QString tryname;
+    QString trysrc;
+    QVector<Domain *> ret;
+    
+    m_conn = virStorageVolGetConnect(m_vol);
+    virDomainPtr *domains;
+    int count = virConnectListAllDomains(m_conn, &domains, VIR_CONNECT_LIST_DOMAINS_ACTIVE | VIR_CONNECT_LIST_DOMAINS_INACTIVE);
```

Comment:
Can't you go up on the parent() and get the Connection class which will give you Domains objects and there you get this, would be less work or less duplicated work.

Reply:
that could be my to less knowledge in c++ .. i was trying to get that with the available routines but didn't get it to work... and at the end i was thinking this is doing less work than your complex routines and was doing it this way. The work was inspired by the virt-manager...

Reply:
ok, I can refactor this later anyway